### PR TITLE
[SourceGen] Diagnostics Cleanup

### LINKED ISSUE #1323: [SourceGen] Cleanup diagnostic descriptors
Cleanup FunctionMetadata generation related diagnostic descriptors and add more tests similar to [this startup extension test](https://github.com/Azure/azure-functions-dotnet-worker/blob/172a1101b5f5940b2a479907c08c8da3c33a496d/test/Sdk.Generator.Tests/ExtensionStartupRunnerGeneratorTests.cs#L170). 

Diagnostic descriptor messages are currently lengthy and difficult to test. 

Items:

- [ ] Clean up source gen descriptors
- [ ] Re-enable `MultipleOutputOnMethodFails` test

### PR DESCRIPTION
**Pull request by satvu**
### Issue describing the changes in this PR

resolves #1323 

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: sdk/Sdk.Generators/DiagnosticDescriptors.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ private static DiagnosticDescriptor Create(string id, string title, string messa
         public static DiagnosticDescriptor MultipleHttpResponseTypes { get; }
                   = Create(id: "AZFW0007",
                     title: "Symbol could not be found in user compilation.",
-                    messageFormat: "Found multiple public properties with type '{0}' defined in output type '{1}'.Only one HTTP response binding type is supported in your return type definition.",
+                    messageFormat: "Found multiple public properties of type HttpResponseData defined in return type '{0}'.Only one HTTP response binding type is supported in your return type definition.",
```
Comment (satvu) [inline sdk/Sdk.Generators/DiagnosticDescriptors.cs:48]: Original error message was taken from [legacy generator](https://github.com/Azure/azure-functions-dotnet-worker/blob/b152392e347acbfd97a55fae7f887e55ed5f3542/sdk/Sdk/FunctionMetadataGenerator.cs#L316). I think it's _unlikely_ for this diagnostic error to be hit, but going for parity with legacy generator on error handling.

Comment (liliankasem) [inline sdk/Sdk.Generators/DiagnosticDescriptors.cs:48]:
```suggestion
                    messageFormat: "Found multiple public properties of type HttpResponseData defined in return type '{0}'. Only one HTTP response binding type is supported in your return type definition.",
```

**Review by liliankasem — APPROVED** [review]

File: test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/DiagnosticResultTests.cs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            [Fact]
+            public async void MultipleOutputBindingsOnMethodFails()
+            {
+                var inputCode = @"using System;
+                using System.Net;
+                using System.Collections;
+                using System.Collections.Generic;
+                using Microsoft.Azure.Functions.Worker;
+                using Microsoft.Azure.Functions.Worker.Http;
+                using System.Linq;
```
Comment (madelinegordon) [inline test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/DiagnosticResultTests.cs:56]: nit - usings ordering here

File: test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/DiagnosticResultTests.cs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                    expectedDiagnosticResults);
+            }
+
+            [Fact]
+            public async void MultipleHttpResponseBindingsFails()
+            {
+                var inputCode = @"using System;
+                using Microsoft.Azure.Functions.Worker;
+                using Microsoft.Azure.Functions.Worker.Http;
+                using System.Threading.Tasks;
```
Comment (madelinegordon) [inline test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/DiagnosticResultTests.cs:154]: nit - usings ordering

**Review by madelinegordon — APPROVED** [review]

**Review by surgupta-msft — APPROVED** [review]

File: test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/IntegratedTriggersAndBindingsTests.cs [hunk 4]
```diff
@@ -1,9 +1,11 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
```
Comment (surgupta-msft) [inline test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/IntegratedTriggersAndBindingsTests.cs:4]: We can remove if they are unused _usings_

**Pull request merged**